src/Compilers/CSharp/Portable/Errors/DiagnosticInfoWithSymbols.cs

```diff
@@ -14,32 +12,13 @@ internal class DiagnosticInfoWithSymbols : DiagnosticInfo
         internal DiagnosticInfoWithSymbols(ErrorCode errorCode, object[] arguments, ImmutableArray<Symbol> symbols)
             : base(CSharp.MessageProvider.Instance, (int)errorCode, arguments)
         {
-#if DEBUG
```

Comment:
Why remove all the debug asserts?

Reply:
Or is there no need to assert that it can be a Symbol or a SymbolWithAnnotations?

---
In reply to: [199582682](https://github.com/dotnet/roslyn/pull/28239#discussion_r199582682) [](ancestors = 199582682)

Reply:
Previously we were asserting the arguments were not `SymbolWithAnnotations`. Now `TypeSymbolWithAnnotations` arguments are allowed. And instances of the other `SymbolWithAnnotations` types are rare and `DiagnosticInfo` already checks arguments are `IFormattable`.